decimal.go

```diff
@@ -99,8 +99,9 @@ var (
 	ErrEmptyString = fmt.Errorf("parse empty string")
 
 	// ErrMaxStrLen is returned when the input string exceeds the maximum length
-	// This limitation is set to prevent large string input which can cause performance issue
-	// Maximum length is set to 200
+	// Maximum length is set to 200 so string length value can fit in 1 byte (for MarshalBinary).
```

Comment:
```suggestion
	// Maximum length is arbitrarily set to 200 so string length value can fit in 1 byte (for MarshalBinary).
```

Reply:
Updated. Although, I would not 100% agree that this is an arbitrary number as there are some reasons to choose it. (a number that doesn't exceeds 1 byte and still big enough)

Reply:
You could use empirically or opionniated then, maybe

Reply:
well, arbirary seems fine to me